server/controllers/notification.js

```diff
@@ -0,0 +1,69 @@
+const asyncHandler = require("express-async-handler");
+const Notification = require("../models/Notification");
+
+//@desc Create notification
+//@route POST /notifications
+//@ access Private
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const { title, description, read } = req.body;
+
+  const notification = await new Notification({
+    title,
+    description,
+    read,
+    userId: req.user.id,
+  }).save();
+
+  res.status(201).json({ success: true, data: notification });
+});
+
+// @desc Fetch all notifications
+// @route GET /notifications
+// @access Private
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+  const notifications = await Notification.find({ userId: req.user.id });
+  res
+    .status(200)
+    .json({ success: true, data: notifications, count: notifications.length });
+});
+
+// @desc Fetch all unread notifications
+// @route GET /notifications/unread
+// @access Private
+exports.getUnreadNotifications = asyncHandler(async (req, res, next) => {
+  const unreadNotifications = await Notification.find({ read: { $eq: false } });
+  res.status(200).json({
+    success: true,
+    data: unreadNotifications,
+    count: unreadNotifications.length,
+  });
+});
+
+// @desc Mark notifications as read
+// @route PATCH /notifications/read/:notificationId
+// @access Private
+exports.markNotificationAsRead = asyncHandler(async (req, res, next) => {
+  const notification = await Profile.findById(req.params.notificationId);
+
+  if (!notification) {
+    res.status(404);
+    throw new Error(`No notification with id ${req.params.notificationId}`);
+  }
+
+  const unreadNotifications = await Notification.find({ read: { $eq: false } });
+
+  if (!unreadNotifications) {
+    res.status(400);
+    throw new Error("Bad request");
+  }
+
+  const markAsRead = await Notification.findByIdAndUpdate(
+    req.params.notificationId,
+    { read: req.body.read },
+    {
+      new: true,
+    }
+  );
```

Comment:
A couple of issues here:
- We don't need to `find` the notification again as we already `find` it at `line 46`. We can simply update the document in plain js and use `.save()` on the document.
- If we are marking a notification as read, then we don't need to receive a `read` param in the request body. We can just set `read` to `true`.